modules/apps/translation/translation-web/src/main/resources/META-INF/resources/js/translate/Translate.js

```diff
@@ -30,14 +30,49 @@ const ACTION_TYPES = {
 	UPDATE_FIELDS_BULK: 'UPDATE_FIELDS_BULK',
 };
 
+const getInfoFields = (infoFieldSetEntries = []) => {
+	const sourceFields = {};
+	const targetFields = {};
+
+	infoFieldSetEntries.forEach(({fields}) => {
+		fields.forEach(({id, sourceContent, targetContent}) => {
+			sourceFields[id] = sourceContent;
+
+			targetFields[id] = {
+				content: targetContent,
+				message: '',
+				status: '',
+			};
+		});
+	});
+
+	return {
+		sourceFields,
+		targetFields,
+	};
+};
+
+const normalizeFields = (fields = []) =>
```

Comment:
`normalize` transforms the collection it receives from the API and `denormalize` transforms the collection it will send to the API. Do you have a better name in mind?

Reply:
`normalizeFields` -> `massageResponse`, `massageAutoTranslateResponse`, something like that.

`denormalizeFields' -> `prepareFields`. Something that indicates better what is being done.